menas/src/main/scala/za/co/absa/enceladus/menas/services/DatasetService.scala

```diff
@@ -102,4 +101,150 @@ class DatasetService @Autowired() (datasetMongoRepository: DatasetMongoRepositor
     }
   }
 
+  override def importItem(item: Dataset, username: String): Future[Option[Dataset]] = {
+    getLatestVersionValue(item.name).flatMap {
+      case Some(version) => update(username, item.copy(version = version))
+      case None => super.create(item.copy(version = 1), username)
+    }
+  }
+
+  override def validateSingleImport(item: Dataset, metadata: Map[String, String]): Future[Validation] = {
+    val confRulesWithConnectedEntities = item.conformance.filter(_.hasEntityConnected)
+    val maybeSchema = datasetMongoRepository.getConnectedSchema(item.schemaName, item.schemaVersion)
+
+    val validations = super.validateSingleImport(item, metadata)
+    val validationsWithSchema = validateSchema(item.schemaName, item.schemaVersion, validations, maybeSchema)
+    val validationWithConnected = validateConnectedEntities(confRulesWithConnectedEntities, validationsWithSchema)
+    validateConformanceRules(item.conformance, maybeSchema, validationWithConnected)
+  }
+
+  private def validateConnectedEntities(confRulesWithConnectedEntities: List[ConformanceRule],
+                                    validationWithSchema: Future[Validation]): Future[Validation] = {
+    confRulesWithConnectedEntities.foldLeft(validationWithSchema) { (acc, cr) =>
+      val maybeConnectedEntity = cr.maybeConnectedEntityType.get match {
+        case "mappingTable" => datasetMongoRepository.getConnectedMappingTable(
+          cr.maybeConnectedEntityName.get,
+          cr.maybeConnectedEntityVersion.get
+        )
+      }
+
+      val issueMsg =
+        s"""Connected ${cr.maybeConnectedEntityType.get} ${cr.maybeConnectedEntityName.get}
+           | v${cr.maybeConnectedEntityVersion.get} could not be found""".stripMargin.replaceAll("[\\r\\n]", "")
+
+      for {
+        ce <- maybeConnectedEntity
+        accValidations <- acc
+      } yield accValidations.withErrorIf(ce.isEmpty, s"item.${cr.maybeConnectedEntityType.get}", issueMsg)
```

Comment:
```suggestion
      } yield ce.map{entity=>accValidations.withError(s"item.${cr.maybeConnectedEntityType.get}", issueMsg)}.getOrElse(accValidations)```

I don't see the added value of `withErrorIf` when you have to blindly call `Option.get` (I know that it is "by design" guarded by the `key` and `error` params being `=>` by name, but still, I don't think it read nicely.

Reply:
This is now marked as outdated, but has not been addressed, I think.

Reply:
Correct. I know about it.

Reply:
Fixed